### PR TITLE
CIF-350 - As a developer I want to have the possibility to model MoneyValue and not just price

### PR DESCRIPTION
**Pull request by dplaton**
Replace the `Price` with a more generic `MoneyValue` concept - the `Price` class is renamed to `MoneyValue`. However, the name of the fields which represent a price are NOT changed since they make perfect semantic sense.
The tax-related objects (`TaxInfo` and `TaxPortion`) are now using the `MoneyValue` fields to represent value (instead of the old `BigDecimal`).

### CODE REVIEW

File: src/main/java/com/adobe/commerce/cif/model/common/TaxInfo.java [hunk 1]
```diff
@@ -14,25 +14,24 @@
...
 package com.adobe.commerce.cif.model.common;
 
-import java.math.BigDecimal;
 import java.util.List;
 
 import io.swagger.annotations.ApiModelProperty;
 
 public class TaxInfo {
 
     @ApiModelProperty(value = "The amount in cents for the tax info.", required = true)
```
Comment (masodu) [inline src/main/java/com/adobe/commerce/cif/model/common/TaxInfo.java]: pls update docs to replace amount with value and remove cents.

File: src/main/java/com/adobe/commerce/cif/model/common/TaxPortion.java [hunk 2]
```diff
@@ -24,7 +22,7 @@
     protected String name;
 
     @ApiModelProperty(value = "The amount in cents for the tax portion.", required = true)
-    protected BigDecimal amount;
```
Comment (masodu) [inline src/main/java/com/adobe/commerce/cif/model/common/TaxPortion.java]: pls update docs to replace amount with value and remove cents.

File: src/main/java/com/adobe/commerce/cif/model/discount/Discount.java [hunk 3]
```diff
@@ -32,7 +32,7 @@
     protected String description;
 
     @ApiModelProperty(value = "The amount which is discounted. Subtract this to obtain new price.", required = true)
-    protected Price amount;
```
Comment (masodu) [inline src/main/java/com/adobe/commerce/cif/model/discount/Discount.java]: pls change amount to value

Comment (masodu) [reply src/main/java/com/adobe/commerce/cif/model/discount/Discount.java]: I would also remove: `Subtract this to obtain new price.`

File: src/main/java/com/adobe/commerce/cif/model/cart/ShippingInfo.java [hunk 4]
```diff
@@ -27,10 +27,10 @@
     protected String name;
     
     @ApiModelProperty(value = "The shipping price.", required = true)
-    protected Price price;
```
Comment (mhaack) [inline src/main/java/com/adobe/commerce/cif/model/cart/ShippingInfo.java]: I recommend renaming the field name to `cost`, since shipping fee is a cost you pay, not a price.

File: src/main/java/com/adobe/commerce/cif/model/cart/ShippingInfo.java [hunk 5]
```diff
@@ -27,10 +27,10 @@
...
+    protected MoneyValue price;
     
     @ApiModelProperty(value = "The discounted shipping price.")
-    protected Price discountedPrice;
```
Comment (mhaack) [inline src/main/java/com/adobe/commerce/cif/model/cart/ShippingInfo.java]: See above, rename to `cost`.

File: src/main/java/com/adobe/commerce/cif/model/cart/ShippingMethod.java [hunk 6]
```diff
@@ -29,7 +29,7 @@
     protected String description;
     
     @ApiModelProperty(value = "The price of the shipping method aka shipping cost.", required = true)
-    protected Price price;
```
Comment (mhaack) [inline src/main/java/com/adobe/commerce/cif/model/cart/ShippingMethod.java]: I recommend renaming the field name to `cost`, since shipping fee is a cost you pay, not a price. The comment already uses cost.

File: src/test/resources/cart.json [hunk 7]
```diff
@@ -52,23 +52,39 @@
                 "amount": 10000
             },
             "taxInfo": {
-                "amount": 1000,
+                "value": {
+                  "currency": "USD",
```
Comment (cjelger) [inline src/test/resources/cart.json]: fix 2 spaces formatting, thx

File: src/test/java/com/adobe/commerce/cif/model/PojoTest.java [hunk 8]
```diff
@@ -59,13 +59,17 @@ private SetterGetter(Method setter, Method getter) {
     @Test
     public void testSettersAndGetters() throws Exception {
         Set<ClassInfo> classes = ClassPath.from(this.getClass().getClassLoader()).getTopLevelClassesRecursive(MODEL_PACKAGE);
-        for (Class<?> clazz : classes.stream().map(c -> c.load()).collect(Collectors.toList())) {
-            LOGGER.debug("Checking setter/getter for " + clazz.getName());
```
Comment (cjelger) [inline src/test/java/com/adobe/commerce/cif/model/PojoTest.java]: Why did you remove the debug? I think it's quite useful to make sure that for example we test all the expected classes.

File: src/test/resources/cart.json [hunk 9]
```diff
@@ -52,23 +52,39 @@
...
+                  "amount": "900",
+                  "country":"US"
+                },
                 "portions": [
                     {
                         "name": "VAT 10%",
-                        "amount": 1000
+                        "value": {
+                          "currency": "USD",
```
Comment (cjelger) [inline src/test/resources/cart.json]: more 2 spaces formatting

File: src/test/resources/taxInfo.json [hunk 10]
```diff
@@ -1,9 +1,17 @@
...
-    "portions": [
-        {
-            "name": "VAT 10%",
-            "amount": 900
+    "value": {
+        "currency": "USD",
+        "amount": "900",
+        "country": "US"
+    },
+  "portions": [
```
Comment (cjelger) [inline src/test/resources/taxInfo.json]: alignment/formatting seems wrong ... activate IDE auto-formatting?

File: src/test/java/com/adobe/commerce/cif/model/PojoTest.java (hunk 8 above)
Comment (dplaton) [reply src/test/java/com/adobe/commerce/cif/model/PojoTest.java]: If it's useful then maybe we should promote the message to `info` level?

File: src/test/resources/taxInfo.json (hunk 10 above)
Comment (dplaton) [reply src/test/resources/taxInfo.json]: If I format it via IDE then I get a ton of changes (not sure why).

File: src/test/java/com/adobe/commerce/cif/model/PojoTest.java (hunk 8 above)
Comment (cjelger) [reply src/test/java/com/adobe/commerce/cif/model/PojoTest.java]: Should stay at `debug`, otherwise it's creating too much output, but I activated that many times when developing the test to make sure it really tested everything.

File: src/test/resources/cart.json [hunk 11]
```diff
@@ -52,23 +52,39 @@
                 "amount": 10000
             },
             "taxInfo": {
-                "amount": 1000,
+                "value": {
+                    "currency": "USD",
+                    "amount": "900",
```
Comment (cjelger) [inline src/test/resources/cart.json]: Sorry I initially missed that, all these amounts should be numbers, not strings (there are quite a few in many JSON examples). BTW, you changed the amounts, while I initially made sure the amounts match so that the example makes sense.
